example/include/OMR/App/Object.hpp

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#if !defined(OMR_APP_OBJECT_HPP_)
+#define OMR_APP_OBJECT_HPP_
+
+#include <omrcfg.h>
+
+#include <stdint.h>
+#include <cstdlib>
+
+namespace OMR {
+namespace App {
+
+class ObjectSlotHandle {
+public:
+	ObjectSlotHandle(fomrobject_t* slot) : _slot(slot) {}
+
+	void setGcReference(fomrobject_t value) const {
+		*_slot = value;
+	}
+
+	fomrobject_t getGcReference() const {
+		return *_slot;
+	}
+
+private:
+	fomrobject_t* _slot;
+};
+
+class ObjectSlotIterator {
+public:
+	ObjectSlotIterator& operator++() {
+		++_slot;
+		return *this;
+	}
+
+	ObjectSlotHandle operator*() const {
+		return ObjectSlotHandle(_slot);
+	}
+
+protected:
+	friend class Object;
+
+	explicit ObjectSlotIterator(fomrobject_t* slot) : _slot(slot) {}
+
+private:
+	fomrobject_t* _slot;
+};
+
+typedef uint8_t ObjectFlags;
+
+class ObjectHeader {
+public:
+	explicit ObjectHeader(size_t sizeInBytes, ObjectFlags flags = 0) {
+		assign(sizeInBytes, flags);
+	}
+
+	size_t sizeInBytes() const {
```

Comment:
I'm avoiding the words get/set in accessors. let me know if you want this changed.